jl4-core/src/L4/EvaluateLazy.hs

```diff
@@ -225,8 +226,16 @@ nfDirective (MkEvalDirective r traced expr env) = do
       else fmap (, Nothing) $ tryEval $ do
         whnf <- runConfig $ ForwardMachine env expr
         nf whnf
-  let finalTrace = postprocessTrace <$> mt
-  pure (MkEvalDirectiveResult r v finalTrace)
+  let
+    finalTrace = postprocessTrace <$> mt
+    v' =
+      if isAssert
+        then Assertion
+          case v of
+            Right (MkNF nf') | Just True <- boolView nf' -> True
```

Comment:
you could have replaced this with `Right (MkNF (ValBool True))`, theoretically..

Reply:
Plausible. Nvm then.